master/static/migrations/20240516014542_add-workspace-namespace-bindings-table.tx.up.sql

```diff
@@ -0,0 +1,6 @@
+CREATE TABLE public.workspace_namespace_bindings (
+    workspace_id INT REFERENCES workspaces(id) ON DELETE CASCADE,
+    cluster_name text NOT NULL,
```

Comment:
@salonig23 @stoksc Should we remove the `NOT NULL` constraint here? Since we don't require users to specify a cluster name with regular K8s rm?

Reply:
We just wouldn't add it in the database in that case. the code in pods.go (future jobs.go) will infer that it would either the default namespace specified in master config or "default"

Reply:
Gotchyaa thanks Saloni!

Reply:
Hmm, I think we would still need to add it to the db in case someone wants to change their workspace namespace binding to anything other than the default. 
You may have to check on your end that if we query a workspace-namespace binding with no associated cluster name, then assign the workload to the default (defined in master config or just "default") namespace if the queried namespace doesn't exist in K8s (which can happen if someone redeploys master using a different cluster but supplies no cluster name, which is permitted for single RM k8s clusters).
I have a function `VerifyNamespaceExists` that does this check, so I can make this change on my end after rebasing onto your branch if you want!
however, that would still mean removing the `NOT NULL` constraint from `cluster_name` in the db. WDYT?